glib/src/subclass/types.rs

```diff
@@ -425,6 +450,10 @@ unsafe extern "C" fn instance_init<T: ObjectSubclass>(
     let imp = T::with_class(klass);
 
     ptr::write(imp_storage, imp);
+
+    // Any additional instance initialization.
+    let obj = from_glib_none::<_, Object>(obj.cast());
```

Comment:
```suggestion
    let obj = from_glib_borrow::<_, Object>(obj.cast());
```

I'm surprised that this didn't cause memory leaks or crashes with GTK :) You would be stealing the floating reference here.

Reply:
If I make it borrow then don't I need to store `&T` in `InitializingObject`, thus requiring a lifetime generic parameter? Just wanted to avoid that.

Reply:
No, you would store `Borrow<T>` inside `InitializingObject`.

Reply:
Can you cast `Borrow<T>`? I need to cast it from `Object` to `Self::Type`.

Reply:
Yes, see all the signal handlers: https://github.com/gtk-rs/gtk-rs/blob/f316f4025cde40953f2e47bcd58c8f400ca92bc7/gtk/src/auto/button.rs#L733

Reply:
Doesn't this grab an `&Object` from within `Borrowed<Object>` and cast that into `&T`? While what I'd need is `Borrowed<Object> -> Borrowed<T>`

Reply:
You can do the cast as part of `as_ref()`. Or you can also do `Borrowed::new(Object::from_glib_borrow(ptr).into_inner().unsafe_cast())` if you prefer that.